scripts/con_spec.js

```diff
@@ -0,0 +1,59 @@
+function (context, args) { // T2 CON_SPEC passes in {s:"string", d:"digit"}
+
+//
+//   Author: Pobega
+//   ▄▄▄▄▄▄▄ ▄▄▄ ▄ ▄    ▄  ▄▄  ▄▄▄▄▄▄▄
+//   █ ▄▄▄ █ █▀█▄ ▄ ▄▀▄▀█▄▄█▄█ █ ▄▄▄ █
+//   █ ███ █ ▀▀ ▀ ▄█▄▄▄█▀▄ ▄█▀ █ ███ █
+//   █▄▄▄▄▄█ █ ▄▀█▀█ ▄▀▄▀▄ ▄ █ █▄▄▄▄▄█
+//   ▄▄  ▄▄▄  ▀ ▄▀▄█ ▀█▀ ██▀█▄  ▄ ▄▄▄▄
+//    ▄▄█ ▀▄█▀▀▀▄█▄▄█▄ ▀ ▄▀██ █ ▀▄▄█▄
+//   █ ▀██▄▄▀█ █▀▄█ ▀ █ ▀██▀ ▄▀█ █  ▀▄
+//   ██▀ ▄▄▄ █▄   ▀▀█▀▄█ ▄▀ █▀▀ █ █▄█
+//   ▄▀█  █▄▀███▄▀▄█▀▄█▀█▄█ ▄ █▄▀█  ██
+//   ▄▀ ▀▀▄▄█ ▀▀▄█▄▄█▀ █  ▀ █▀▀ ▀  ▀▄
+//     ▀   ▄▄ █ ▀▄█ █ █▀▀██▄██▄█ ▄  ▀▄
+//   ▀▀ ▀▀▀▄▄▄▄▀  ▀▀█▀▄▀ ▀  █ █▄█▄▀ ▄
+//   ▄▄█▀▄▀▄ ██▀▄▀▄█▄▀███▄█▄ ▄▄▄██  █▀
+//   ▄▄▄▄▄▄▄ ▀▄▄▄█▄▄█▀▀█ ▄█▄▀█ ▄ █▀▀▄
+//   █ ▄▄▄ █ █▀▄▀▄█  ▀█▀▄▀█▄ █▄▄▄██ █▀
+//   █ ███ █  █▀  ▀▀█▀█▀  ▄▀▀▀▀█▀█▄▄
+//   █▄▄▄▄▄█ █▀█▄▀▄█▀▀█  ▄██▄█▄▄▀ ▀  ▄
+//   http://github.com/Pobega/hackmud
+//
+// :::TRUST COMMUNICATION::: {security_level:4, length:132}
+//
+// This script is the answer to the wolf and up CON_SPEC response.
+// Pass it as a scriptor to CON_SPEC to pass the lock.
+//
+// Ex: npc.loc{CON_SPEC:#s.user.con_spec}
+//
+
+    if (args == undefined || args.d.constructor != Number || args.s.constructor != String) {
```

Comment:
checks that args were passed, and that the types are correct on the args. If statements evaluate from left to right, so checking that args is defined happens first and short circuits into the usage page.

Reply:
Actually this fails if you run `con_spec { d:4 }` with "Cannot read property constructor of undefined". I will look at this later.